src/media/js/site/login.js

```diff
@@ -25,15 +27,23 @@ export function loginRequired(Component, LoginHandler, group,
   class AuthenticatedComponent extends React.Component {
     render() {
       const isLoggedIn = !!this.props.user.token;
-      const hasPermission = (process.env.NODE_ENV === 'test' ||
-                             _checkPermissions(this.props.user, group));
 
-      if (isLoggedIn && hasPermission) {
-        return <Component/>;
+      if (isLoggedIn) {
+        const hasPermission = (process.env.NODE_ENV === 'test' ||
+                               _checkPermissions(this.props.user, group));
+        const hasSignedTOS = this.props.user.tos.has_signed;
```

Comment:
might want to check that tos exists. QA will probably be already logged in and the user object will be cached. this will break and QA won't be able to log out without doing ls.clear